### PR TITLE
Grid filtering changes and bugfixes

### PR DESCRIPTION
**Pull request by EmanH**
- Added overloads to getFormatedValueOperation to accept cell coordinates or a column as the first argument
- Added method on grid controller to get the default date format
- Added an internal property to TransformPipeline to track transforms and exposed hasTransform() to check if the transform is already a part of the pipeline 
- Fixed the filterRowsOperation so it actually works with the row transform pipeline and it filters on formatted values now
- Changed GridSelectionRange to accept grid controller instead of grid events in its constructor
- Added a method to grid data source to upsert rows for handling delta changes
- Added dropdownMenu parameters to IGridColumnInterface and implementation
- Fixed a bug in the body component where records would not show if they were provided before the component was initialized
- Optimized cell component detectChanges() to only receive the value into the cell type if it is different from the currrent value
- Now cloning the original data source in SingleSelectGridDropdownOverlayComponent as we need the row pipeline to be distinct
- Added grid dropdown menu to the HeaderComponent

### CODE REVIEW

File: projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts [hunk 1]
```diff
@@ -9,14 +9,67 @@ export class GetFormattedValue extends Operation {
...
   constructor(factory: ICellOperationFactory) { super(factory.gridController) }
 
-  public run(coords: IGridCellCoordinates, value?: any, returnHtml = true): string {
-    value = value ?? this.cellOperations.GetCellValue.run(coords)?.value
-    if (value === null || value === undefined) return ''
-    const cellType = this.cellOperations.GetCellType.run(coords)
-    switch (cellType.name) {
+  public getPlainText(coords: IGridCellCoordinates, value?: any): string
+  public getPlainText(column: IGridColumn, value?: any): string
+  public getPlainText(arg1: IGridCellCoordinates | IGridColumn, value?: any): string
```
Comment (cli2022) [inline projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts:14]: `arg1` could be renamed to `coords`

Comment (EmanH) [reply projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts]: I've applied these changes in the latest commit

File: projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts [hunk 2]
```diff
@@ -9,14 +9,67 @@ export class GetFormattedValue extends Operation {
...
+  {
+    return this._handleOverload(arg1, value, false)
+  }
+
+  public getHTML(coords: IGridCellCoordinates, value?: any): string
+  public getHTML(column: IGridColumn, value?: any): string
+  public getHTML(arg1: IGridCellCoordinates | IGridColumn, value?: any): string
```
[the remark this answers is not in the document]
Comment (EmanH) [reply projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts]: I've applied these changes in the latest commit

File: projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts [hunk 3]
```diff
@@ -9,14 +9,67 @@ export class GetFormattedValue extends Operation {
...
+      return this._run(input, value, html)
+    } else if (this._isCoordinates(input)) {
+      return this._run(input, value, html)
+    }
+    throw new Error('Invalid argument')
+  }
+
+  private _run(coords: IGridCellCoordinates, value: any, returnHtml: boolean): string
+  private _run(column: IGridColumn, value: any, returnHtml: boolean): string
+  private _run(arg1: IGridCellCoordinates | IGridColumn, value: any, returnHtml: boolean): string
```
[the remark this answers is not in the document]
Comment (EmanH) [reply projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts]: I've applied these changes in the latest commit

File: projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts [hunk 4]
```diff
@@ -9,14 +9,67 @@ export class GetFormattedValue extends Operation {
...
+  }
+
+  public getHTML(coords: IGridCellCoordinates, value?: any): string
+  public getHTML(column: IGridColumn, value?: any): string
+  public getHTML(arg1: IGridCellCoordinates | IGridColumn, value?: any): string
+  {
+    return this._handleOverload(arg1, value, true)
+  }
+
+  private _isCoordinates(arg: IGridCellCoordinates | IGridColumn): arg is IGridCellCoordinates {
```
[the remark this answers is not in the document]
Comment (EmanH) [reply projects/ngx-grid-core/src/lib/controller/cell-operations/get-formated-value.operation.ts]: I've applied these changes in the latest commit

**Review by cli2022 — APPROVED** [review]
@EmanH The changes look good.

**Pull request merged**